### PR TITLE
Removed CalibTracker/SiStripCommon dependency from CalibFormats/SiStripObjects

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

CalibFormat objects should not be reading files to get intermediate data as this violates one of the guiding principles of CMSSW.  This change moves the needed information into a separate class, SiStripDetInfo and requires code that constructs a SiStripQuality or a SiStripGain to pass in the appropriate SiStripDetInfo.

The class SiStripDetInfoFileReader now returns the SiStripDetInfo.

This now makes CalibFormats/SiStripObjects independent from algorithmic code.

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@davidlange6 this should help with the ROOT modules build.

**Comment by jfernan2** [comment]
please test

**Comment by Dr15Jones** [comment]
@smuzaffar why did the `<use name="dd4hep"/>` in Geometry/TrackerNumberingBuilder/BuildFile.xml not get applied to the plugins in CalibTracker/SiStripESProducers?

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
I have another round of cleanups on its way.

**Comment by Dr15Jones** [comment]
finished with what I wanted to do

**Comment by Dr15Jones** [comment]
Please test

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
NOTE: ` test SiStripDAQ_O2O_test had ERRORS` is failing in the IBs as well. Therefore it is not a consequence of this PR.

**Comment by mmusich** [comment]
@cmsbuild, please test.
In the last IB the issue with `SiStripDAQ_O2O_test` [is solved](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_12_0_X_2021-06-02-1100/unitTestLogs/CondTools/SiStrip#/)

**Comment by jfernan2** [comment]
+1

File: RecoLocalTracker/SiStripZeroSuppression/BuildFile.xml [hunk 1]
```diff
@@ -8,6 +8,7 @@
 <use name="CondFormats/SiStripObjects"/>
 <use name="CalibFormats/SiStripObjects"/>
 <use name="CalibTracker/Records"/>
+<use name="Geometry/TrackerGeometryBuilder"/>
```
Comment (slava77) [inline RecoLocalTracker/SiStripZeroSuppression/BuildFile.xml:11]: why was this needed?
There are no includes of `Geometry/TrackerGeometryBuilder` in ` RecoLocalTracker/SiStripZeroSuppression/`

Comment (slava77) [reply RecoLocalTracker/SiStripZeroSuppression/BuildFile.xml:11]: nevermind, I removed the comment; there's no problem here

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/33888	f589717
- code changes in reco are rather trivial: 
    - adapting to the new interface in RecoLocalTracker/SiStripClusterizer/test
    - update/fix of some missing package dependencies
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by civanch** [comment]
+1

**Comment by yuanchao** [comment]
+1

* remove CalibTracker/SiStripCommon dependency
* matrix tests and unit tests passed

**Comment by qliphy** [comment]
kindly ping @cms-sw/db-l2

**Comment by Dr15Jones** [comment]
Ping @cms-sw/db-l2

**Comment by ggovi** [comment]
+1

**Comment by qliphy** [comment]
+1
